Source/WarpX.H

```diff
@@ -121,11 +122,28 @@ public:
 
     void Evolve (int numsteps = -1);
 
-    void EvolveImplicitPicardInit (int lev);
-    void SaveParticlesAtImplicitStepStart (WarpXParticleContainer& pc, int lev);
-    void FinishImplicitParticleUpdate (WarpXParticleContainer& pc, int lev);
-    void FinishImplicitFieldUpdate(amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > >& Efield_fp,
-                                   amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > >& Efield_n);
+    //
+    // Functions used by implicit solvers
+    //
+    void PreRHSOp ( const amrex::Real  cur_time,
+                    const amrex::Real  a_full_dt,
+                    const int          a_nl_iter,
+                    const bool         a_from_jacobian );
+    void SaveParticlesAtImplicitStepStart ( );
+    void FinishImplicitParticleUpdate ( );
+    void UpdateElectricField( const WarpXSolverVec&  a_E, const bool a_apply );
```

Comment:
Should these functions have more specific names? e.g. `SetElectricFieldAndApplyBCs`